Core/MMKV_IO.cpp

```diff
@@ -645,6 +655,25 @@ bool MMKV::setDataForKey(MMBuffer &&data, MMKVKey_t key, bool isDataHolder) {
         if (itr != m_dic->end()) {
             bool onlyOneKey = !m_isInterProcess && m_dic->size() == 1;
             if (likely(!m_enableKeyExpire)) {
+
+                // compare data before appending to file
+                if (m_enableCompareBeforeSet) {
+                    MMBuffer rawData = getDataForKey(key);
+                    if (isDataHolder) {
```

Comment:
The `isDataHolder` has something to do with the param passed in. The relationship with the data already inside MMKV is unclear & confusing if any.

Reply:
An simple assumption can be made: for a same TYPE of key,  set and get must be equivalent.
that's to say,:
when setting for a type with isDataHolder = true, getting its value must read a variant length for this holder.
when setting for a type with isDataHolder = false, getting its value need not read a variant length for any holder.

Anyway your comment is valuable, is there any better idea for handing this two cases?

Thanks

Reply:
It's possible for the user to store a **different kind** of value for the same key, even though not recommended.

Reply:
got it

Reply:
Let's consider this:
```
    mmkv1->set("value1", "key1");
    printf("%d\n", mmkv1->getBool("key1", false)); // print 1
    printf("actualSize = %lu\n", mmkv1->actualSize()); // print actualSize = 26
    mmkv1->set(true, "key1");
    printf("%d\n", mmkv1->getBool("key1", false)); // print 1
    printf("actualSize = %lu\n", mmkv1->actualSize()); // print actualSize = 33，** it is wrong **
    mmkv1->set("value1", "key1");
    string v;
    mmkv1->getString("key1", v);
    printf("%s\n", v.c_str()); // print value1
```

Is it your meaning?

Reply:
The bottom line is that we can't make any assumption on the type of existing value according to the param passed in (aka `isDataHolder`).

Reply:
I have been stunned by TYPE modification.

Reply:
JUST want to make sure that, does the new method work?

Reply:
No. It desn't.

Reply:
Some safety check are needed for string、char*、MMBuffer、vector<string>

Will this ways be OK for  bool、int、int32、int64、uint64、float、double？ absolutely, Excluding reuse iterator to avoid two find of unordered_map.